include/stdexec/__detail/__type_traits.hpp

```diff
@@ -30,8 +30,14 @@ namespace stdexec {
   // __decay_t: An efficient implementation for std::decay
 #if STDEXEC_HAS_BUILTIN(__decay)
 
+  namespace __tt {
+    template <class _Ty>
+    struct __decay {
+      using type = __decay(_Ty);
+    };
+  } // namespace __tt
```

Comment:
the whole purpose of the intrinsic is to avoid the cost of instantiating a class template. this should achieve the same goal with only a single class template instantiation:

```suggestion
  namespace __tt {
    template <bool>
    struct __decay_ {
      template <class _Ty>
      using __f = __decay(_Ty);
    };
  } // namespace __tt
```